### PR TITLE
Browser bookmark

### LINKED ISSUE #4072: Possibility of "bookmarking" URLs
image

Would be nice to be able to bookmark URLs and have them show up in the "Applications" page (under their own "Web-based dapps" heading). Page's title can be used for the dapp's name and URL can be used for the sub-heading. Version and author probably can't be shown.

### PR DESCRIPTION
**Pull request by CraigglesO**
closes #4072

Adds bookmarks to the Applications page:

alt-1

Feel free to edit the icon image.

Late night on caffeine, lost creativity for "Web Browser Link"... suggestions?

### CODE REVIEW

File: js/src/views/Web/store.js [hunk 1]
```diff
@@ -30,16 +31,20 @@ let instance = null;
...
   @observable parsedUrl = null;
   @observable nextUrl = null;
   @observable token = null;
 
   constructor (api) {
     this._api = api;
 
+    this.favorites = localStore.get(FAVORITES_STORE) || {};
     this.nextUrl = this.currentUrl = this.loadLastUrl();
+    this.isFavorited = this.favorites[this.currentUrl];
```
Comment (tomusdrw) [inline js/src/views/Web/store.js]: `isFavorited` should be computable, right now it's not updated when `this.currentUrl` changes (unless it's being changed by `setCurrentUrl`)

File: js/src/views/Dapps/dappsStore.js [hunk 2]
```diff
@@ -255,7 +268,27 @@ export default class DappsStore extends EventEmitter {
   }
 
   @action readDisplayApps = () => {
+    let browserFavorites = store.get(FAVORITES_STORE) || {};
+
+    browserFavorites = Object.keys(browserFavorites).map((fav) => {
+      return {
+        url: 'web',
+        author: '',
+        name: fav.replace('https://', ''),
```
Comment (tomusdrw) [inline js/src/views/Dapps/dappsStore.js]: What about `http://`? Also do we need to do it twice?

File: js/src/views/Web/store.js (hunk 1 above)
Comment (CraigglesO) [inline js/src/views/Web/store.js]: I originally did computed. It wasn't refreshing -_-
I always have 50/50 luck with computed and mobx.
It updates upon construction and each new request to the browser. I couldn't find a fail case.

File: js/src/views/Dapps/dappsStore.js (hunk 2 above)
Comment (CraigglesO) [inline js/src/views/Dapps/dappsStore.js]: I could refactor both those issues no problem.

**Comment by CraigglesO** [comment]
I had no problem opening any url with the browser. The point was to give a quick shot to any browser link via the applications page per Gavin's request. I didn't show the screenshot, but you click the "star" (bookmark) in the browser, and it's added to applications page. In reality, the way I am using this is way more beneficial when ui-2 comes out.

**Comment by gavofyork** [comment]
would very much prefer to open in new page/tab when chrome + extension is available.

**Pull request closed without merging**